lib/galaxy/managers/history_contents.py

```diff
@@ -617,6 +593,40 @@ def get_filter(attr, op, val):
             return self.parsed_filter(filter_type="orm", filter=column_filter)
         return super()._parse_orm_filter(attr, op, val)
 
+    def get_query_filters_with_relations(self, query_filters: ValueFilterQueryParams, related_q: str, history_id):
+        """Return `query_filters_with_relations` changing `related:hid` to `related:[hid1, hid2, ...]`."""
+        if query_filters.q and query_filters.qv:
+            qv_index = query_filters.q.index(related_q)
+            qv_hid = query_filters.qv[qv_index]
+
+            # Type check whether hid is int
+            if not qv_hid.isdigit():
```

Comment:
Ideally we check inputs as early as possible. The q/qv syntax makes it hard to do that with pydantic, but I think it should move back up to parse_query_filters_with_relations

Reply:
Done, moved back to `parse_query_filters_with_relations`